Classes/IndexQueue/RecordMonitor.php

```diff
@@ -173,21 +64,16 @@ protected function getUpdateSubPagesRecursiveTriggerConfiguration()
      * @param string $command The command.
      * @param string $table The table the record belongs to
      * @param int $uid The record's uid
-     * @param string $value
-     * @param DataHandler $tceMain TYPO3 Core Engine parent object
      */
     public function processCmdmap_preProcess(
         $command,
         $table,
-        $uid,
-        /** @noinspection PhpUnusedParameterInspection */
-        $value,
-        DataHandler $tceMain
-    ) {
+        $uid
+    ): void {
         if ($command === 'delete' && $table === 'tt_content' && $GLOBALS['BE_USER']->workspace == 0) {
-            // skip workspaces: index only LIVE workspace
-            $pid = $this->getValidatedPid($tceMain, $table, $uid);
-            $this->indexQueue->updateItem('pages', $pid, time());
+            $this->eventDispatcher->dispatch(
+                new ContentElementDeletedEvent((int)$uid)
```

Comment:
Do we not lose the PID there?
Or we collect the PID via soft deleted records?

Reply:
Right, there is no need to validate and store the pid for tt_content records here, as we know that tt_content uses soft deletes and we can do that processing downstream.

The used `DataHandler` method `getPID` that is used to determine and validate the pid, ignores the `DeletedRestriction`.